src/ScriptEditor/NetscriptDefinitions.d.ts

```diff
@@ -3914,11 +3919,44 @@ export interface CodingContract {
    * Note that this is not the same as the contract’s description.
    * This is just the data that the contract wants you to act on in order to solve the contract.
    *
+   * There is an optional third parameter where you can assert the contract type.
+   * This will type the data returned for that specific contract type.
+   * Be aware that this function will throw if the type in the fourth parameter is not the same as the type of the contract with that filename.
+   *
    * @param filename - Filename of the contract.
    * @param host - Host of the server containing the contract. Optional. Defaults to current server if not provided.
    * @returns The specified contract’s data, data type depends on contract type.
    */
-  getData(filename: string, host?: string): CodingContractData;
+  getData(filename: string, host?: string): any;
+  getData<T extends CodingContractName | `${CodingContractName}`>(
```

Comment:
I think the extra overhead of the duplicate signature and duplicate docs is just not worth it here. If you want TS typesafety, use `getContract`.

Reply:
I answered this above in the discussion about the `attempt` signature.